snuba/state/rate_limit.py

```diff
@@ -166,60 +166,16 @@ def _get_bucket_key(prefix: str, bucket: str, shard_id: int) -> str:
     return "{}{}{}".format(prefix, bucket, shard_suffix)
 
 
-@contextmanager
-def rate_limit(
+def rate_limit_start_request(
     rate_limit_params: RateLimitParameters,
-) -> Iterator[Optional[RateLimitStats]]:
-    """
-    A context manager for rate limiting that allows for limiting based on:
-        * a rolling-window per-second rate
-        * the number of queries concurrently running.
-
-    It uses one redis sorted set to keep track of both of these limits
-    The following mapping is kept in redis:
-
-        bucket: SortedSet([(timestamp1, query_id1), (timestamp2, query_id2) ...])
-
-
-    Queries are thrown ahead in time when they start so we can count them
-    as concurrent, and thrown back to their start time once they finish so
-    we can count them towards the historical rate. See the comments for
-    an example.
-
-               time >>----->
-    +-----------------------------+--------------------------------+
-    | historical query window     | currently executing queries    |
-    +-----------------------------+--------------------------------+
-                                  ^
-                                 now
-    """
-    (bypass_rate_limit, rate_history_s, rate_limit_shard_factor,) = state.get_configs(
-        [
-            # bool (0/1) flag to disable rate limits altogether
-            ("bypass_rate_limit", 0),
-            # number of seconds the timestamps are kept
-            ("rate_history_sec", 3600),
-            # number of shards that each redis set is supposed to have.
-            # increasing this value multiplies the number of redis keys by that
-            # factor, and (on average) reduces the size of each redis set
-            ("rate_limit_shard_factor", 1),
-        ]
-    )
-    assert isinstance(rate_history_s, (int, float))
-    assert isinstance(rate_limit_shard_factor, int)
-    assert rate_limit_shard_factor > 0
-
-    if bypass_rate_limit == 1:
-        yield None
-        return
-
+    query_id: str,
+    rate_history_sec,
+    rate_limit_shard_factor,
+) -> RateLimitStats:
     now = time.time()
 
-    query_id_uuid = uuid.uuid4()
-    query_id = str(query_id_uuid)
-
     # Compute the set shard to which we should add and remove the query_id
-    bucket_shard = int(query_id_uuid) % rate_limit_shard_factor
+    bucket_shard = hash(query_id) % rate_limit_shard_factor
```

Comment:
this change is necessary to allow any arbitrary string to be used as the query id. If people feel really strongly about this I can change it but I would also have to change the allocation policy to require UUIDs as well and I don't think there's a meaningful difference